src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3491,10 +3488,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           vparams.lengthIs <= MaxFunctionArity &&
           (formal baseType FunctionClass(vparams.length) match {
             case TypeRef(_, _, formalargs) =>
-              ( exists2(formalargs, vparams)((formal, vparam) =>
-                        vparam.tpt.isEmpty && (tparams exists formal.contains))
-                && (tparams forall isLowerBounded)
-              )
+              exists2(formalargs, vparams) { (formal, vparam) =>
+                vparam.tpt.isEmpty && tparams.exists(formal.contains)
+              } && tparams.forall(isLowerBounded)
```

Comment:
Without trying to understand it, maybe:
```
case (formal, Function(vparams, _)) if vparams.exists(_.tpt.isEmpty) && vparams.lengthIs <= MaxFunctionArity && tparams.forall(isLowerBounded) =>
  formal.baseType(FunctionClass(vparams.length)) match {
    case TypeRef(_, _, formalargs) => exists2(formalargs, vparams)((formal, vparam) => vparam.tpt.isEmpty && tparams.exists(formal.contains))
    case _ => false
  }
case _ => false
```

Reply:
Now is it better or worse to first get the length of `vparams`?

Reply:
Or it just doesn't matter...

Reply:
Oh `tparams` can be extracted even further up

Reply:
I'm resisting understanding the code, but I bet it doesn't matter. :)

Reply:
I bet the condition `<= MaxFunctionArity` is true most of the time so better to get the length first. Splitting hairs here 😄